jaeger-core/src/main/java/io/jaegertracing/JaegerTracer.java

```diff
@@ -168,13 +168,15 @@ public ScopeManager scopeManager() {
   }
 
   @Override
-  public Span activeSpan() {
+  public JaegerSpan activeSpan() {
```

Comment:
I think this will need to remain Span, to allow other Spans to be stored in the `ScopeManager` - e.g. java-metrics (using api-extensions lib). Don't think this should be a problem - we can focus on returning Jaeger specific implementations based on the fluent api - from build span -> start. It would be the apps responsibility to then return the Jaeger impl if they needed to continue using it

Reply:
Do you have an example that I could test this? I could then add a new test based on the example.

Reply:
I think the following test should be sufficient:

```java
    // prepare
    Span activeSpan = mock(Span.class);
    ScopeManager scopeManager = tracer.scopeManager();

    // test
    scopeManager.activate(activeSpan, false);

    // check
    assertEquals(activeSpan, tracer.activeSpan());
```

Reply:
+1